### PR TITLE
core/includes: add common macros file

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR wants to make common macros available, first example is `XTSTR`, `esp32` already uses it in some assembler instructions but it can also be useful for compile time error messages based on a macros value, e.g:

```c
#pragma message "The value for X is out of range, X < " XSTR(X_MAX_VALUE)
```

I'm unsure about the proper location for this kind of macros, but it somehow didn't make sense to put it in `kernel_defines.h`, it didn't seem to fit with the macros defined there, although with the current description in `kernel_defines.h`, the line between what should or should not be there is unclear to me.

### Testing procedure

- Green Murdock for esp32.

- Does the naming for the file make sense? How to limit what goes in?

### CODE REVIEW

**Comment by kaspar030** [comment]
>     * Does the naming for the file make sense? How to limit what goes in?

I'd prefer `include/macros/xtstr.h` (one macro per file). Answers the second question nicely.

**Comment by fjmolinas** [comment]
> >     * Does the naming for the file make sense? How to limit what goes in?
> 
> I'd prefer `include/macros/xtstr.h` (one macro per file). Answers the second question nicely.

Done

File: core/include/macros/xtstr.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_internal
+ * @{
+ *
+ * @file
+ * @brief       Common macros
```
Comment (leandrolanzieri) [inline core/include/macros/xtstr.h:14]: I guess this description should change now?

Comment (fjmolinas) [reply core/include/macros/xtstr.h]: I added them to a different group as well let me know if that makes sense

**Comment by fjmolinas** [comment]
This could also be in sys maybe?  since in https://github.com/RIOT-OS/RIOT/pull/13530 it was proposed to add them there.

File: core/include/macros/xtstr.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @def     XTSTR
+ *
+ * @brief   A function to return the string respresentation of x
```
Comment (bergzand) [inline core/include/macros/xtstr.h:31]:
```suggestion
 * @brief   A macro to return the string respresentation of x
```

**Comment by bergzand** [comment]
> This could also be in sys maybe? since in #13530 it was proposed to add them there.

I think core is fine here. We could add a separate `sys/includes/macros` for the more specific (or less core-y) macros

**Comment by bergzand** [comment]
Please squash!

File: core/include/macros/xtstr.h (hunk 1 above)
Comment (leandrolanzieri) [reply core/include/macros/xtstr.h]: Thanks!

**Pull request merged**